ui/src/dashboards/selectors/index.test.ts

```diff
@@ -18,34 +24,99 @@ const untypedGetTimeRangeByDashboardID = getTimeRange as (a: {
+  ranges: RangeState
+  currentDashboard: CurrentDashboardState
+  app: AppPresentationState
+}) => TimeRange
+
 describe('Dashboards.Selector', () => {
-  const dashboardIDs = ['04c6f3976f4b8001', '04c6f3976f4b8000']
+  const dashboardIDs = [
+    '04c6f3976f4b8001',
+    '04c6f3976f4b8000',
+    '04c6f3976f4b8002',
+  ]
+  const lower = '2020-05-05T10:00:00-07:00'
+  const upper = '2020-05-05T11:00:00-07:00'
+  const customTimeRange = {
+    lower: moment(lower).toISOString(),
+    upper: moment(upper).toISOString(),
+    type: 'custom',
+  } as CustomTimeRange
   const ranges: RangeState = {
     [dashboardIDs[0]]: pastFifteenMinTimeRange,
     [dashboardIDs[1]]: pastHourTimeRange,
+    [dashboardIDs[2]]: customTimeRange,
   }
 
-  it('should return the the correct range when a matching dashboard ID is found', () => {
+  it('should return the correct range when a matching dashboard ID is found', () => {
     const currentDashboard = {id: dashboardIDs[0]}
 
     expect(
       untypedGetTimeRangeByDashboardID({ranges, currentDashboard})
     ).toEqual(pastFifteenMinTimeRange)
   })
 
-  it('should return the the default range when no matching dashboard ID is found', () => {
+  it('should return the default range when no matching dashboard ID is found', () => {
     const currentDashboard = {id: 'Oogum Boogum'}
 
     expect(
       untypedGetTimeRangeByDashboardID({ranges, currentDashboard})
     ).toEqual(DEFAULT_TIME_RANGE)
   })
 
-  it('should return the the default range when no ranges are passed in', () => {
+  it('should return the default range when no ranges are passed in', () => {
     const currentDashboard = {id: dashboardIDs[0]}
 
     expect(
       untypedGetTimeRangeByDashboardID({ranges: {}, currentDashboard})
     ).toEqual(DEFAULT_TIME_RANGE)
   })
+
+  it('should return the an unmodified version of the timeRange when the timeZone is local', () => {
+    const currentDashboard = {id: dashboardIDs[2]}
+    const app: AppPresentationState = {
+      ephemeral: {
+        inPresentationMode: false,
+      },
+      persisted: {
+        autoRefresh: 0,
+        showTemplateControlBar: false,
+        navBarState: 'expanded',
+        timeZone: 'Local' as TimeZone,
+        theme: 'dark',
+      },
+    }
+
+    expect(
+      untypedGetTimeRangeWithTimeZone({ranges, currentDashboard, app})
+    ).toEqual(customTimeRange)
+  })
+
+  it('should return the timeRange for the same hour with a UTC timezone when the timeZone is UTC', () => {
+    const currentDashboard = {id: dashboardIDs[2]}
+
+    const app: AppPresentationState = {
+      ephemeral: {
+        inPresentationMode: false,
+      },
+      persisted: {
+        autoRefresh: 0,
+        showTemplateControlBar: false,
+        navBarState: 'expanded',
+        timeZone: 'UTC' as TimeZone,
+        theme: 'dark',
+      },
+    }
+
+    const expected = {
+      lower: setTimeToUTC(lower),
+      upper: setTimeToUTC(upper),
+      type: 'custom',
```

Comment:
Rather than hardcoding the values as I was doing, I set the time more deterministically using the setTimeToUTC function.

Reply:
i'm not sure this test is actually testing much at this point. when you strip everything away it basically says

```ts
assertEquals(setTimeToUTC(lower), setTimeToUTC(lower))
```

since [this line](https://github.com/influxdata/influxdb/pull/18011/files#diff-be9be89f3c9a10c0102cf057f74e8104R25) is the code under test.

don't think we should remove it, but i do think you should spend some time thinking about ways you can get more confidence from this test.

Reply:
I totally agree. The only other implementation of this that I could think of would be to calculate the UTCOffset locally and set a value dynamically through a switch statement with all potential values